.travis.yml

```diff
@@ -0,0 +1,65 @@
+language: cpp
+
+sudo: required
+
+dist: trusty
+
+compiler:
+  - gcc
+
+env:
+#  global:
+#    - INSTALL_DIR=~/mylibs
+  CUDA=7.0-28
+
+script:
+  - ls -l
+  - mkdir build_tmp && cd build_tmp
+  - cmake ..
+  - make check_all VERBOSE=1
+
+before_script:
+#  - cat /etc/apt/sources.list
+#  - cat /etc/apt/sources.list.d/*
+  - cat /etc/apt/sources.list
+  - uname -a
+#  - apt-cache search boost
+#  - apt-cache search nvidia
+#  - sudo apt-add-repository multiverse
+#  - sudo apt-get update -qq
+#  - sudo dpkg --configure -a
+#  - sudo apt-get install -f -qq
+#  - sudo dpkg --get-selections | grep hold || { echo "All packages OK."; }
+  - sudo apt-get install -q -y cmake-data cmake
+  - cmake --version
+  
+  - sudo apt-get install -qq build-essential
+  - gcc --version && g++ --version # 4.8
+#  - sudo apt-get install -qq nvidia-common
+#  - sudo apt-get install -qq nvidia-cuda-dev nvidia-cuda-toolkit # 5.5
+  
+  - travis_retry wget http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1404/x86_64/cuda-repo-ubuntu1404_${CUDA}_amd64.deb
```

Comment:
i do understand that caching will stuff persistent between builds. but in the example everthing is built from source. is that somehow the point, do we regard it as better practice?